### PR TITLE
Fix colorized semantic segmentation normalization

### PR DESCRIPTION
**Pull request by AntoineRichard**
# Description

Fixes `mdp.image(normalize=True)` returning colorized semantic-segmentation observations as `uint8`. The normalizer now recognizes the renderer contract for this output (RGBA `uint8`) and returns the same normalized `float32` representation used for RGB-like images. Raw `int32` semantic-ID maps remain unchanged.

No new dependencies.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots

Not applicable.

## Checklist

- [x] I have read and understood the contribution guidelines
- [x] I have run the pre-commit checks with `uv run isaaclab -f`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by kellyguo11 — APPROVED** [review]

**Review by mmichelis — COMMENTED** [review]
In practice the normalize might be called on warp arrays as well, could they? These changes currently only work for torch tensors, is that intended?

**Comment by AntoineRichard** [comment]
> In practice the normalize might be called on warp arrays as well, could they? These changes currently only work for torch tensors, is that intended?

No that function clearly looks torch only for now:

```python
    if is_rgb_like(data_type):
    if is_rgb_like(data_type) or (data_type == "semantic_segmentation" and images.dtype == torch.uint8):
        if images.dtype == torch.uint8 and images.ndim == 4 and images.is_contiguous():
            return normalize_image_uint8(images, channel_dim=channel_dim, out=out)
        # PyTorch fallback for callers that pre-floated or pass a strided view.
        resolved_channel_dim = channel_dim + images.ndim if channel_dim < 0 else channel_dim
        spatial_dims = tuple(d for d in range(1, images.ndim) if d != resolved_channel_dim)
        images = images.float() / 255.0
        images -= torch.mean(images, dim=spatial_dims, keepdim=True)
        return images
    if is_depth_like(data_type):
        images[images == float("inf")] = 0
        return images
    if is_normals_like(data_type):
        return (images + 1.0) * 0.5
    return images
```

**Review by mmichelis — APPROVED** [review]
